vmm/src/lib.rs

```diff
@@ -215,20 +293,32 @@ impl Vmm {
 
             if let Some(ref vm_config) = self.vm_config {
                 let vm = Vm::new(Arc::clone(vm_config), exit_evt, reset_evt)?;
-                self.vm = Some(vm);
+                self.vm = Some(Arc::new(Mutex::new(vm)));
+                self.migration_register().expect("Fail to register migration state component");
             }
         }
 
         // Now we can boot the VM.
         if let Some(ref mut vm) = self.vm {
-            vm.boot()
+            let mut vm = vm.lock().unwrap();
+            vm.boot()?;
+
+            let migration = self.migration.clone();
+            thread::Builder::new()
+                .name(format!("migration_state_get"))
+                .spawn(move || {
+                    migration.take_snapshot();
+                });
```

Comment:
Why are we asking for a VM snapshot when booting the device? The snapshot trigger should come from an HTTP request, translated into an API command sent to the VMM, which would then trigger the VMM to build a snapshot.

Reply:
This is a "test" patch. It is just used to show the state register/get routes are workable.

The first two patches are to provide the migration state handling infrastructure. For real work, we need implement in next phase.

Reply:
Ok. I think we should remove that part, to make it something mergeable.

Reply:
Yes, this patch will not present in formal version. The purpose of it is to make you easily test the patches.